content/general/events.textile

```diff
@@ -72,9 +72,276 @@ h3(#sources). Sources
+- data := an optional data payload for the published message. Find out more about "message data":/rest/messages#data
+- encoding := if the data payload is present, this encoding specifies the format of the data. If missing or empty, then the data payload is a string type. For example, if JSON data is set for the published message's "@data@":/rest/messages#data attribute, then the @encoding@ would be "@json/utf-8@"
+- id := unique ID assigned by Ably to this message
+- timestamp := the time in milliseconds since the epoch when this message was received by Ably
+
+h5(#decoding-messages). Decoding messages
+
+Each member of @data.messages@ is a 'raw' message. Messages sent "over the realtime service":/realtime/channels are automatically decoded into "@Messages@":/realtime/types#message object by the Ably client library. With webhooks you need to to do this explicitly, using "@Message.fromEncodedArray@":/realtime/messages#message-from-encoded-array on the @data.messages@ array, or "@Message.fromEncoded@":/realtime/messages#message-from-encoded or an individual member of that array. This will transform them into an array of "@Messages@":/realtime/types#message (or in the case of @fromEncoded@, an individual "@Message@":/realtime/types#message). This has several advantages, e.g.:
+
+* It will fully decode any @data@ (using the @encoding@) back into the same datatype that it was sent in (or an equivalent in each client library's language)
+* If you are using "encryption":/realtime/encryption, you can pass your encryption key to the method and it will decrypt the @data@ for you
+
+We recommend you do this for all messages you receive over webhooks. For example (using ably-js):
+
+```[javascript]
+webhookMessage.items.forEach((item) => {
+  const messages = Ably.Realtime.Message.fromEncodedArray(item.data.messages);
+  messages.forEach((message) => {
+    console.log(message.toString());
+  })
+})
+```
+
+h6(#example-channel-message). Example @channel.message@ JSON payload
+
+```[json]
+{
+   "items":[
+      {
+         "webhookId":"d28HJw",
+         "source":"channel.message",
+         "serial":"e91URQe6QA0gwm05496618:2",
+         "timestamp":1479301189860,
+         "name":"channel.message",
+         "data":{
+            "channelId":"foob",
+            "site":"eu-central-1-A",
+            "messages":[
+               {
+                  "id":"9qaOH1C4tO:2:0",
+                  "name":"foo",
+                  "connectionId":"9qaOH1C4tO",
+                  "timestamp":1479301189856,
+                  "data":"bar"
+               }
+            ]
+         }
+      }
+   ]
+}
+```
+
+Please note that if you are planning to receive messages via WebHooks, it is theoretically very easy to exceed the transport rate limits we impose on WebHooks to prevent DoS attacks against your own servers. We recommend you consider "message queues":/general/queues instead as they are far more scalable.
```

Comment:
Following a rearchitecting in how webhooks work, we're no longer pushing people away from webhooks quite so hard as that -- iirc the limits are the same as for our internal queues

Reply:
Thanks, removing.